cmake/winml.cmake

```diff
@@ -618,3 +618,12 @@ option(onnxruntime_BUILD_WINML_TESTS "Build WinML tests" ON)
 if (onnxruntime_BUILD_WINML_TESTS)
   include(winml_unittests.cmake)
 endif()
+
+# This is needed to suppress warnings that complain that no imports are found for the delayloaded library cublas64*.lib
+# When cuda is enabled in the pipeline, it sets CMAKE_SHARED_LINKER_FLAGS which affects all targets including winml_dll.
+# However, there are no cuda imports in winml_dll, and the linkler throws the 4199 warning.
+# This is needed to allow winml_dll build with cuda enabled.
+set_target_properties(winml_dll
+  PROPERTIES
+  LINK_FLAGS
+  "/ignore:4199")
```

Comment:
>/ignore:4199 [](start = 3, length = 12)

do we care about other dlls that might have this issue? this will ignore all of them right?

Reply:
probably - but it looks like this problem is systemic in the ort repo because they modify global cmake vars

---
In reply to: [372584575](https://github.com/microsoft/onnxruntime/pull/2936#discussion_r372584575) [](ancestors = 372584575)